Source/WebKit/WPEPlatform/wpe/wayland/WPEViewWayland.cpp

```diff
@@ -284,6 +322,26 @@ const struct wl_callback_listener frameListener = {
     }
 };
 
+const struct zwp_linux_buffer_release_v1_listener bufferReleaseListener = {
+    // fenced_release
+    [](void* userData, struct zwp_linux_buffer_release_v1*, int32_t fence)
+    {
+        auto* buffer = WPE_BUFFER(userData);
+        wpe_buffer_dma_buf_set_release_fence(WPE_BUFFER_DMA_BUF(buffer), fence);
+        wpe_view_buffer_released(wpe_buffer_get_view(buffer), buffer);
+        if (auto* dmaBufBuffer = static_cast<DMABufBuffer*>(wpe_buffer_get_user_data(buffer)))
+            g_clear_pointer(&dmaBufBuffer->release, zwp_linux_buffer_release_v1_destroy);
+    },
+    // immediate_release
+    [](void* userData, struct zwp_linux_buffer_release_v1*)
+    {
+        auto* buffer = WPE_BUFFER(userData);
+        wpe_view_buffer_released(wpe_buffer_get_view(buffer), buffer);
+        if (auto* dmaBufBuffer = static_cast<DMABufBuffer*>(wpe_buffer_get_user_data(buffer)))
+            g_clear_pointer(&dmaBufBuffer->release, zwp_linux_buffer_release_v1_destroy);
```

Comment:
This is duplicated code we could move it to a helper and call it from both release callbacks. For example, `wpe_view_wayland_buffer_released`